bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.insteon.internal.handler;
+
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.insteon.internal.InsteonBinding;
+import org.openhab.binding.insteon.internal.config.InsteonNetworkConfiguration;
+import org.openhab.binding.insteon.internal.discovery.InsteonDeviceDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link InsteonNetworkHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Rob Nielsen - Initial contribution
+ */
+@NonNullByDefault
+public class InsteonNetworkHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(InsteonNetworkHandler.class);
+
+    private @Nullable InsteonNetworkConfiguration config;
+    private @Nullable InsteonBinding insteonBinding;
+    private @Nullable InsteonDeviceDiscoveryService insteonDeviceDiscoveryService;
+    private @Nullable ScheduledFuture<?> pollingJob = null;
+
+    public static ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+
+    public InsteonNetworkHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Starting Insteon bridge");
+        config = getConfigAs(InsteonNetworkConfiguration.class);
+
+        scheduler.execute(() -> {
+            insteonBinding = new InsteonBinding(this, config.getPort(), config.getDeviceStatisticsInterval(),
+                    config.getDeviceDeadCount(), config.getDevicePollInterval(), config.getAdditionalDevices(),
+                    config.getAdditionalFeatures(), config.getModemDbRetryTimeout());
+
+            updateStatus(ThingStatus.ONLINE);
+
+            if (!insteonBinding.startPollingWhenReady()) {
+                String msg = "Initialization failed, unable to start the Insteon bridge with the port '"
+                        + config.getPort() + "'.";
+                logger.error(msg);
```

Comment:
There is a lot of logging to error in the code. Logging to error should only be done if it's a fatal bug. For example here it should log to debug as already the status is set. Can you check all loggings and change it to the appropriate level. See also https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I will review and remove the error logging.